NOTIF-103 Prepare pages for disabled/enabled permissions.

josejulio opened the pull request:
Currently using the "notifications" appId, need to update later
 to use integrations on it's page.

### Without write permission:
![Screenshot_2021-01-11_10-48-18](https://user-images.githubusercontent.com/3845764/104212705-1dfdc880-53fb-11eb-8697-83b4bd728f9a.png)
![Screenshot_2021-01-11_10-47-54](https://user-images.githubusercontent.com/3845764/104212711-1fc78c00-53fb-11eb-98b2-6d7d3cfb974c.png)

#### Without read permission
![Screenshot_2021-01-11_10-55-35](https://user-images.githubusercontent.com/3845764/104213095-91073f00-53fb-11eb-9c93-19d43df59912.png)
![Screenshot_2021-01-11_10-48-54](https://user-images.githubusercontent.com/3845764/104212734-26560380-53fb-11eb-8dcc-d4ca09a461ad.png)


#### With Read & Write permissions
![Screenshot_2021-01-11_10-50-22](https://user-images.githubusercontent.com/3845764/104212758-2e15a800-53fb-11eb-84d0-1d0b8b67af31.png)
![Screenshot_2021-01-11_10-49-54](https://user-images.githubusercontent.com/3845764/104212761-2eae3e80-53fb-11eb-8a33-c1a1c29ecba3.png)


Once we have the RBAC permissions set we need to update the way we fetch the RBAC. Currently only fetching `notifications`.

```diff
diff --git a/src/app/useApp.ts b/src/app/useApp.ts
index e72d844..9927cba 100644
--- a/src/app/useApp.ts
+++ b/src/app/useApp.ts
@@ -28,6 +28,8 @@ export const useApp = (): Omit<AppContext, 'rbac' | 'applications'> & Partial<Pi
                 case Config.integrations.subAppId:
                     document.title = Config.integrations.title;
                     break;
+                default:
+                    throw new Error(`Invalid app id found for this application ${appId}`);
             }
 
             insights.chrome.identifyApp(appId);
@@ -36,8 +38,9 @@ export const useApp = (): Omit<AppContext, 'rbac' | 'applications'> & Partial<Pi
 
     useEffect(() => {
         waitForInsights().then(insights => {
+            const appId = insights.chrome.getApp();
             insights.chrome.auth.getUser().then(() => {
-                fetchRBAC(Config.appId).then(setRbac);
+                fetchRBAC(appId).then(setRbac);
             });
         });
     }, []);
diff --git a/src/config/Config.ts b/src/config/Config.ts
index 165a7b3..6d9492e 100644
--- a/src/config/Config.ts
+++ b/src/config/Config.ts
@@ -6,7 +6,6 @@ const apiBaseUrl = `/api/notifications/${apiVersion}`;
 export const withBaseUrl = (path: string) => `${apiBaseUrl}${path}`;
 
 const Config = {
-    appId: 'notifications',
     integrations: {
         subAppId: 'integrations',
         title: 'Integrations | Settings'

```

adamruzicka commented on src/SmartComponents/NotificationEdit/NotificationEdit.js:
```diff
@@ -155,7 +155,14 @@ export class NotificationEdit extends Component {
         .catch(() => {
-            const errors = _.mapValues(this.props.endpointErrors.errors, (value) => ({ errors: value }));
+            let errors = {};
+            _.forEach(this.props.endpointErrors, (value) => {
```
When I was testing this, no matter what I tried `this.props.endpointErrors` was always undefined.

bastilian commented:
@adamruzicka Mind taking another look.

Validation errors are now passed on correctly and are marking the fields as they should.
There is also now the option to ignore 404 errors for filters to not show an error at all.

When the backend is down the proxy seems to still respond with 200, and and empty response, I'm unsure how to identify those properly as errors with the backend to show a fitting notification.

adamruzicka commented:
Thanks @bastilian !

josejulio committed:
POL-103 Prepare pages for disabled/enabled permissions.
 Currently using the "notifications" appId, need to update later
 to use integrations on it's page.

josejulio committed:
lint: Update import order

josejulio committed:
Remove some warns/errors that were polluting the test output

josejulio committed:
Add testing for notifications page elements disabled without read permissions

josejulio committed:
Test integration page elements that are meant to be disabled with RBAC

AllenBW reviewed (commented):
Does the export action on either of these tables need rbac? 🤔   ALSO does there need to be any indicator sayin why folks dont have write all? or that the action is disabled cuz missing rbac? 

That aside looking good! 🥰

josejulio commented:
> Does the export action on either of these tables need rbac?

Not as far as I'm aware. Read permissions are more than enough to be able to use them, and the whole page requires read permissions, so all is set for that.

> ALSO does there need to be any indicator sayin why folks dont have write all? or that the action is disabled cuz missing rbac?

Is RBAC related permission.

Thanks!
